### PR TITLE
ICU-21349 Add extra ComplexUnitsConverter constructor that takes only CLDR units Identifier

### PR DESCRIPTION
**Pull request by younies**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-21349
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [x] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

**Review by sffc — APPROVED** [review]

File: icu4c/source/test/intltest/units_test.cpp [hunk 1]
```diff
@@ -621,15 +621,10 @@ void UnitsTest::testComplexUnitsConverter() {
     MeasureUnit input, output;
     MeasureUnitImpl tempInput, tempOutput;
     MaybeStackVector<Measure> measures;
-    for (const TestCase &testCase : testCases) {
-        input = MeasureUnit::forIdentifier(testCase.input, status);
-        output = MeasureUnit::forIdentifier(testCase.output, status);
-        const MeasureUnitImpl& inputImpl = MeasureUnitImpl::forMeasureUnit(input, tempInput, status);
-        const MeasureUnitImpl& outputImpl = MeasureUnitImpl::forMeasureUnit(output, tempOutput, status);
-        auto converter = ComplexUnitsConverter(inputImpl, outputImpl, rates, status);
+    auto testATestCase = [&](const ComplexUnitsConverter& converter ,StringPiece initMsg , const TestCase &testCase) {
```
Comment (hugovdm) [inline icu4c/source/test/intltest/units_test.cpp:624]: Formatting.

File: icu4c/source/test/intltest/units_test.cpp [hunk 2]
```diff
@@ -650,7 +645,24 @@ void UnitsTest::testComplexUnitsConverter() {
             assertEquals(msg.data(), testCase.expected[i].getUnit().getIdentifier(),
                          measures[i]->getUnit().getIdentifier());
         }
+    };
+
+    for (const auto &testCase : testCases)
+    {
```
Comment (hugovdm) [inline icu4c/source/test/intltest/units_test.cpp:651]: ICU formatting style has the brace on the same line as the for.

File: icu4j/main/tests/core/src/com/ibm/icu/dev/test/impl/UnitsTest.java [hunk 3]
```diff
@@ -55,7 +55,31 @@ public void testComplexUnitsConverter() {
...
+
+            void testATestCase(ComplexUnitsConverter converter) {
+                List<Measure> measures = converter.convert(value, null).measures;
+
+                assertEquals("measures length", expected.length, measures.size());
+                int i = 0;
+                for (Measure measure : measures) {
+                    double accuracy = 0.0;
+                    if (i == expected.length - 1) {
+                        accuracy = accuracy;
```
Comment (hugovdm) [inline icu4j/main/tests/core/src/com/ibm/icu/dev/test/impl/UnitsTest.java:67]: What is this supposed to do?

**Review by hugovdm — COMMENTED** [review]
Late review, FWIW.

(I try to keep my "review-requested:hugovdm" list short, including closed PRs. That means I either review, or unassign myself.)

**Pull request merged**